Source/santad/EventProviders/EndpointSecurityTestUtil.mm

```diff
@@ -205,6 +205,24 @@ - (void)newClient:(es_client_t *_Nullable *_Nonnull)client
   [self.clients addObject:mockClient];
 }
 
+- (BOOL)removeClient:(es_client_t *_Nonnull) client {
+  MockESClient *clientToRemove = nil;
+  for (MockESClient *c in self.clients) {
+      if (client == (__bridge es_client_t *)c) {
```

Comment:
Use isEqual: instead of ==. While the latter _might_ work OK in this case, the default NSObject impl. of isEqual: definitely will and it can be overridden.

Reply:
In this case I was explicitly looking for pointer equality and wasn't sure if isEqual would be more inclusive.

Reply:
The default implementation is "equal identity". If there's a reason to care that the object in question is at the same address as before then == is appropriate but warrants a comment about _why_ that's what's desired.

Reply:
Comment added.